### PR TITLE
fix: [CDS-69376]: improved error message for GCR and ECR connectors

### PR DESCRIPTION
**Pull request by sahithikolichala**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

#### [Contributor license agreement](https://github.com/harness/harness-core-ui/blob/develop/CONTRIBUTOR_LI
Screen Shot 2023-06-01 at 4 59 52 PM
CENSE_AGREEMENT.md)

### CODE REVIEW

**Comment by sahithikolichala** [comment]
retrigger titlecheck

**Review by cjlee01 — APPROVED** [review]

**Comment by cjlee01** [comment]
![alt screenshot]
(https://i.imgur.com/WdBMIcZ.png "pr creation")

File: src/modules/75-ce/components/Connectors/AWSCOConnector/AWSCOConnector.tsx [hunk 1]
```diff
@@ -34,6 +34,7 @@ const AWSCOConnector: React.FC<COAWSConnectorProps> = props => {
   const { getString } = useStrings()
   return (
     <section className={css.wrapper}>
+      Step1
```
Comment (pankajsingh-harness) [inline src/modules/75-ce/components/Connectors/AWSCOConnector/AWSCOConnector.tsx:37]: This need to be removed

File: src/modules/35-connectors/common/ConnectorTestConnection/ConnectorTestConnection.tsx [hunk 2]
```diff
@@ -342,12 +342,14 @@ const ConnectorTestConnection: React.FC<StepProps<VerifyOutOfClusterStepProps> &
...
       )
       const permissionLink = getPermissionsLink()
+
       return (
         <Layout.Vertical className={css.stepError}>
           {responseMessages ? (
             <ErrorHandler
               responseMessages={responseMessages}
               className={css.errorHandler}
+              url={props?.prevStepData?.dockerRegistryUrl}
```
Comment (pankajsingh-harness) [inline src/modules/35-connectors/common/ConnectorTestConnection/ConnectorTestConnection.tsx:352]: Is `dockerRegistryUrl` present for all the connectors? There is not such prop in the interface. This is not complaining as the interface is generic object. There is `url` prop in the `ConnectorTestConnectionProps` interface. Can we use that?

**Review by pankajsingh-harness — CHANGES_REQUESTED** [review]

**Review by cjlee01 — APPROVED** [review]

**Review by tan-nhu — APPROVED** [review]

**Review by jenil-harness — APPROVED** [review]

**Review by swarajcheguri-harness — APPROVED** [review]

**Review by pankajsingh-harness — APPROVED** [review]

**Pull request merged**